app/src/main/java/protect/card_locker/preferences/Settings.java

```diff
@@ -92,6 +100,22 @@ public String getColor() {
         return getString(R.string.setting_key_theme_color, mContext.getResources().getString(R.string.settings_key_system_theme));
     }
 
+    public int getPreferredColumnCount() {
+        var orientation = mContext.getResources().getConfiguration().orientation;
+        var defaultColumnCount = orientation == ORIENTATION_PORTRAIT ? 2 : 3;
```

Comment:
I thought about the duplication of these values between Settings and the pref `defaultValue` attrs, but if you make them xml values you have to duplicate them between `column_count_values` and `@string/setting_key_default_column_count_$orient` instead, so that hardly seems better

Reply:
I disagree about this change. Because you're changing the logic.

The old logic was as follows:
By default, use 2 columns (https://github.com/CatimaLoyalty/Android/blob/main/app/src/main/res/values/integers.xml)
If the screen width is at least 600dp, use 3 colums (https://github.com/CatimaLoyalty/Android/blob/main/app/src/main/res/values-w600dp/integers.xml)

For most devices, this tends to roughly match portrait vs. landscape. However, it's not consistently the same. For example, [square phones exist](https://github.com/CatimaLoyalty/Android/issues/1339) and Android flipphones with a small secondary screen (which would probably run Catima very well!) are existing too now. Certain tablets and PC devices may, even in landscape, still be at least 600dp.

Can you move it out of the values file to code? Perhaps, although I don't see a benefit from it, the XML is a good location for it. Can you remove the "Automatic" setting and just hardcode it on 2 or 3? I don't think so, not without breaking some use cases.